technobly/vextreme
# implement better brightness control, fixes #46

### Problem

See issue #46 
 
### Solution

- Brightness control was changed to not scale the pixel information, but instead to utilize the global 5-bit brightness control of the APA102-2020 super leds :D
- Brightness RPC function ID 8 was added to allow control of brightness from Vectrex games
- LED Test v2.0 was added to show the new brightness control in action.  See the readme.md in the led-test2/ folder
- System Options structure was added for hw_ver, and rgb_type.  Currently these are used to allow the user to build the STM32 firmware for different LED options, but will also be used for v0.30 VEXTREME hardware which will only have 4 LEDs.

### Steps to Test

- Build the veccart for the different 3 different RGB_TYPE's and led-test2 app, then run the led-test2 app to observe the effects.  Yes I did this :D 

### References

fixes #46
